### PR TITLE
Add GPU.sb3 extension

### PR DESCRIPTION
**Pull request by derpygamer2142**
This adds blocks to use compute shaders in a Turbowarp project. This uses WebGPU(which is experimental!), and only supports compute shaders. From what little research I've done it seems like it would be a pain to add any of the other types, so I'm not doing that.

- Documentation [here](https://extensions.derpygamer2142.com/docs/gpusb3)
- Demos [here](https://extensions.derpygamer2142.com/demos)
- WGSL spec [here](https://www.w3.org/TR/WGSL/)
- WebGPU spec [here](https://www.w3.org/TR/webgpu/)
- WebGPU implementation status by browser [here](https://webgpu.io/)
- WebGPU fundamentals [here](https://webgpufundamentals.org/)

Notes:
- The only thing that's somewhat undocumented(I think, let me know if I forgot something) is buffer alignment and stuff, but webgpufundamentals does a better job of that.
- This doesn't work in the packager with electron. Not sure why, a quick google search leads me to [believe it's in beta](https://github.com/electron/electron/issues/26944).

gpusb3

### CODE REVIEW

**Comment by JeremyGamer13** [comment]
if it doesnt work in electron i would probably add a message stating something like "this wont work with the electron packager options" to the unstable message
also i think the extension thumbnail is fun but might not be good to have on an MIT licensed content only repo, could just be blank until we make one

**Comment by derpygamer2142** [comment]
how about this?
gpusb3 v2

**Review by JeremyGamer13 — CHANGES_REQUESTED** [review]

**Comment by JeremyGamer13** [comment]
looks good but i forgot to click review changes mb

File: static/extensions/derpygamer2142/gpusb3.js [hunk 1]
```diff
@@ -0,0 +1,4485 @@
+// Name: GPU.sb3
+// ID: gpusb3
+// Description: Use WebGPU compute shaders to accelerate your projects.
+// By: derpygamer2142 <https://scratch.mit.edu/users/insanetaco2000/>
+// License: MPL-2.0
```
[the remark this answers is not in the document]
Comment (derpygamer2142) [reply static/extensions/derpygamer2142/gpusb3.js]: it's dual licensed under mit and mpl2.0 now, i think that should be fine

[the remark this answers is not in the document]
Comment (Ianyourgod) [reply static/extensions/derpygamer2142/gpusb3.js]: it should say that then

[the remark this answers is not in the document]
Comment (derpygamer2142) [reply static/extensions/derpygamer2142/gpusb3.js]: oh okay, added

**Review by JeremyGamer13 — APPROVED** [review]

**Comment by JeremyGamer13** [comment]
theres not really a way i could review the whole code within a reasonable time-frame (i know nothing about WebGL or WebGPU) to make sure there are no issues, so beyond my basic checking this seems fine enough to merge

**Pull request merged**